src/Automation/OutputFileHelper.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Axe.Windows.Abstractions;
+using System;
+using Path = System.IO.Path;
+
+namespace Axe.Windows.Automation
+{
+    internal class OutputFileHelper : IOutputFileHelper
+    {
+        private readonly string _outputDirectory;
+        private readonly ISystemDateTime _dateTime;
+
+        public const string DefaultOutputDirectoryName = "AxeWindowsOutputFiles";
+        public const string DefaultFileNameBase = "AxeWindows";
+
+        public OutputFileHelper(string outputDirectory, ISystemFactory systemFactory)
+        {
+            if (systemFactory == null) throw new ArgumentNullException(nameof(systemFactory));
+
+            _dateTime = systemFactory.CreateSystemDateTime();
+            if (_dateTime == null) throw new InvalidOperationException($"Expected {nameof(_dateTime)} not to be null");
+
+            var environment = systemFactory.CreateSystemEnvironment();
+            if (environment == null) throw new InvalidOperationException($"Expected {nameof(environment)} not to be null");
+
+            var directory = systemFactory.CreateSystemIODirectory();
+            if (directory == null) throw new InvalidOperationException($"Expected {nameof(directory)} not to be null");
+
+            if (!String.IsNullOrWhiteSpace(outputDirectory))
```

Comment:
>S [](start = 17, length = 1)

Other places in this file use string with a lower case s. Should we use lower case here for consistency?